cluster-autoscaler/core/scale_down_test.go

```diff
@@ -334,9 +334,169 @@ func TestFindUnneededNodePool(t *testing.T) {
 	assert.NotEmpty(t, sd.unneededNodes)
 }
 
+func TestDeleteNode(t *testing.T) {
+	// common parameters
+	nothingReturned := "Nothing returned"
+	nodeDeleteFailedFunc :=
+		func(string, string) error {
+			return fmt.Errorf("won't remove node")
+		}
+	podNotFoundFunc :=
+		func(action core.Action) (bool, runtime.Object, error) {
+			return true, nil, errors.NewNotFound(apiv1.Resource("pod"), "whatever")
+		}
+
+	// scenarios
+	testScenarios := []struct {
+		name              string
+		pods              []string
+		drainSuccess      bool
+		nodeDeleteSuccess bool
+		expectedDeletion  bool
+	}{
+		{
+			name:              "successful attempt to delete node with pods",
+			pods:              []string{"p1", "p2"},
+			drainSuccess:      true,
+			nodeDeleteSuccess: true,
+			expectedDeletion:  true,
+		},
+		/* Temporarily disabled as it takes several minutes due to hardcoded timeout.
```

Comment:
Leaving this commented out for now as discussed offline with @mwielgus. We're too close to release to refactor now and this scenario increases the time unit tests for this package take from ~1s to ~2 min.